blockchain/blockchain_test.go

```diff
@@ -561,3 +562,48 @@ func TestBlockchainWriteBody(t *testing.T) {
 	fmt.Println(body)
 	fmt.Println(ok)
 }
+
+func TestCalculateGasLimit(t *testing.T) {
+	tests := []struct {
+		name             string
+		blockGasTarget   uint64
+		parentGasLimit   uint64
+		expectedGasLimit uint64
+	}{
+		{
+			name:             "should increase next gas limit towards target",
+			blockGasTarget:   25000000,
+			parentGasLimit:   20000000,
+			expectedGasLimit: 20000000/1024 + 20000000,
+		},
+		{
+			name:             "should decrease next gas limit towards target",
+			blockGasTarget:   25000000,
+			parentGasLimit:   26000000,
+			expectedGasLimit: 26000000 - 26000000/1024,
+		},
+		{
+			name:             "should not alter gas limit",
+			blockGasTarget:   25000000,
+			parentGasLimit:   25000000,
+			expectedGasLimit: 25000000,
+		},
```

Comment:
you are missing a test case where the diff between block gas limits is less than the delta

Reply:
Added the 2 missing test cases, good catch 👍 

fe278e5